### ISSUE TITLE
Help with `[DOTX008E][ERROR] File ... does not exist or cannot be loaded.`

### ISSUE THREAD

**Issue by nfriedly** [dita-ot/dita-ot#1938]
Hi, I'm trying to use DITA-OT 2.0.1 on files that were produced with IBM's ID workbench tool. 

I believe I have DITA installed correctly, it works as expected on the sample files, but it always gives me the same error for any file from the ID workbench tool.

I've checked everything I could think of: file permissions, paths, I put the DTD in a few different places, but nothing seems to make a difference (except mangling the DOCTYPE tag - that gets me a different error.)

Can you give me any guidance on what I'm missing? I'm pretty new to DITA in general, so it might be something obvious.

Here's the file:

```xml
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE map PUBLIC "-//IBM//DTD DITA IBM Map//EN"  "ibm-map.dtd">
<map xml:lang="en-us" chunk="to-content">
    <title>Watson Speech Service for WEA and the Speech Service for WDC</title>
    <topicref href="../wea/topics/wea_tools_wds.dita" navtitle="Enabling the Watson Speech Service">
        <topicref href="overview.ditamap" format="ditamap" navtitle="Overview" otherprops="ctp-only"/>
        <topicref href="layout.ditamap" format="ditamap" navtitle="Layout"/>
        <topicref href="tutorial.ditamap" format="ditamap"/>
        <topicref href="admin.ditamap" format="ditamap" navtitle="Administration" product="wea"/>
        <topicref href="dialog.ditamap" format="ditamap" navtitle="Dialogs"/>
    </topicref>
</map>

```

and, here's the results I get:

    dita -f html5 -i watson_dialog_service.ditamap -o test
    Error: Failed to run pipeline: [DOTX008E][ERROR] File 'file:/Users/nfriedly/Desktop/watson_dialog_service_dita/watson_dialog_service.ditamap' does not exist or cannot be loaded.

I tried making all kinds of tweaks to the file, the only one that got a different result was when I deleted "ibm-map.dtd" from the doctype tag, then I got the following error:

    Error: Failed to run pipeline: [DOTJ012F][FATAL] Failed to parse the input file 'file:/Users/nfriedly/Desktop/watson_dialog_service_dita/watson_dialog_service.ditamap'.: file:/Users/nfriedly/Desktop/watson_dialog_service_dita/watson_dialog_service.ditamap Line 2:The system identifier must begin with either a single or double quote character.

**Comment by easirois** [comment]
The doctype in the map is the IBM specific one and DITA OT doesn't know about those out-of-the-box.  You want to make sure to use the OASIS the doctypes. 


     On Wednesday, June 3, 2015 3:07 PM, Nathan Friedly <notifications@github.com> wrote:
   

 Hi, I'm trying to use DITA-OT 2.0.1 on files that were produced with IBM's ID workbench tool. I believe I have DITA installed correctly, it works as expected on sample files. I've checked everything I could think of: file permissions, paths, I put the DTD in a few different places, but nothing seems to make a difference (except mangling the DOCTYPE tag - that gets me a different error)Can you give me any guidance on what I'm missing? I'm pretty new to DITA in general, so it might be something obvious.Here's the file:<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE map PUBLIC "-//IBM//DTD DITA IBM Map//EN"  "ibm-map.dtd">
<map xml:lang="en-us" chunk="to-content">
    <title>Watson Dialog Service for WEA and the Dialog Service for WDC</title>
    <topicref href="../wea/topics/wea_tools_wds.dita" navtitle="Enabling the Watson Dialog Service">
        <topicref href="overview.ditamap" format="ditamap" navtitle="Overview" otherprops="ctp-only"/>
        <topicref href="layout.ditamap" format="ditamap" navtitle="Layout"/>
        <topicref href="tutorial.ditamap" format="ditamap"/>
        <topicref href="admin.ditamap" format="ditamap" navtitle="Administration" product="wea"/>
        <topicref href="dialog.ditamap" format="ditamap" navtitle="Dialogs"/>
    </topicref>
</map>
and, here's the results I get:dita -f html5 -i watson_dialog_service.ditamap -o test
Error: Failed to run pipeline: [DOTX008E][ERROR] File 'file:/Users/nfriedly/Desktop/watson_dialog_service_dita/watson_dialog_service.ditamap' does not exist or cannot be loaded.
I tried making all kinds of tweaks to the file, the only one that got a different result was when I deleted "ibm-map.dtd" from the doctype tag, then I got the following error:Error: Failed to run pipeline: [DOTJ012F][FATAL] Failed to parse the input file 'file:/Users/nfriedly/Desktop/watson_dialog_service_dita/watson_dialog_service.ditamap'.: file:/Users/nfriedly/Desktop/watson_dialog_service_dita/watson_dialog_service.ditamap Line 2:The system identifier must begin with either a single or double quote character.

**Comment by nfriedly** [comment]
Ok, thanks for the quick response. Any chance of updating that error message to be more specific?

**Comment by easirois** [comment]
That said, if the docs are created by IBMers, but delivered externally, they should be using the OASIS doctypes instead of the IBM specific ones to create the docs.  If you have access to ID workbench, then you can add a reference to the catalog file that IDWB uses into the DITA Open Toolkit catalogue template file.  You want to put it in the template file so it always stays there when/if intergrator task is run in the OT to integrate plugins. 


     On Wednesday, June 3, 2015 3:15 PM, Nathan Friedly <notifications@github.com> wrote:
   

 Ok, thanks for the quick response. Any chance of updating that error message to be more specific?—

**Comment by nfriedly** [comment]
These are internal docs, the external delivery is just HTML (which I'm supposed to create.) 

I don't have a copy of the ID workbench right now, but I'm looking into getting one or at least ta copy of the catalog file. Can you give me any pointers on what that file might look like and how to add the reference once I have it?

**Comment by easirois** [comment]
You can actually download IDWB from http://help.eclipse.org/luna/index.jsp?topic=%2Forg.eclipse.platform.doc.isv%2Fguide%2Fua_help_setup_preferences.htm (still have the link from my IBM days) and there should be a link to IDWB installer from there.  You don't need a license to run the command line tools. You only need the license for the XML editors. Though it a good thing to get the license.  Within that same KC instance you will have have some documentation on the IDWB command line features to process the DITA content. It's not in a really obvious spot if you're just starting out.  map2xhtml is what you need to look for. You will need to install open source version of ObjectRexx (There's a link in the docs as well, but it takes you to the g2O internal wensite, but it's not easy to find the downlaod link from there) Open Object Rexx .  If you need some more help, try contacting my former co-worker in DB2 ID land, Arun Lakhana/Toronto/IBM.  He can point you in the right direction.
|   |
|   |   |   |   |   |
| Open Object RexxHome Welcome Welcome to the Open Object Rexx (ooRexx) web site. This is the home for all the products and projects associated with ooRexx.  |
|  |
| View on www.oorexx.org | Preview by Yahoo |
|  |
|   |


IDWB should get installed in C:\DITATools\IDWB. I believe the catalog file is called idbw-catalog.xml  That said, if you're going to deliver stuff to KC or as a product you should use IDWB since it includes the IBM specific css file for DITA and has the options on the XHTML output to produce a plugin for KC or IEHS.
If you still want to you the OT, then you can update catalog-dita_template.xml. At bottom of the file you can use nextCatalog element to point to the idwb-catalog.xml file.
<nextCatalog catalog="file://C:/DITAtools/IDWB/idwb-catalog.xml"/>  - You will need to test this since I'm running on memory even though it's only been a few months. :)

Hope this helps.
Eric  


     On Wednesday, June 3, 2015 4:41 PM, Nathan Friedly <notifications@github.com> wrote:
   

 These are internal docs, the external delivery is just HTML (which I'm supposed to create.) I don't have a copy of the ID workbench right now, but I'm looking into getting one or at least ta copy of the catalog file. Can you give me any pointers on what that file might look like and how to add the reference once I have it?—

**Comment by juliovaz** [comment]
I have the same issue with a client specialization. The catalog appears to point to the plugin catalog for the doctypes but for some reason it doesn't seem to resolve.Here's a screencap of the catalog in oXygen.

![catalog](https://cloud.githubusercontent.com/assets/7313801/7990177/2f4174ee-0abd-11e5-8234-9cf0f4b6001e.png)

**Comment by robander** [comment]
@nfriedly - if you contact me internally (Robert D Anderson in Rochester, according to Notes), I have a plug-in that supports the IBM Doctypes using DITA-OT. I've got it working on my system right now with DITA-OT 2.0. DITA-OT itself doesn't ship with native support for corporate doctypes, but ours (and others) work fine with a plug-in.

@juliovaz - I've got the IBM Doctypes working fine with a plugin, so it isn't a general issue with all doctype plugins. Your `plugin.xml` file should have something like this (don't think the type=file is needed anymore but I still have it):
```
<feature extension="dita.specialization.catalog.relative" value="catalog-dita.xml" type="file" /> 
```

Your plug-in should also have a the matching catalog file (`catalog-dita.xml` if using the sample above). Inside that catalog, you should have valid relative paths to your doctypes.

**Comment by juliovaz** [comment]
Thanks Robert. I'll look into it when I get back on my work machine. Weird
that problem didn't show up on 1.8.5. Message doesn't help.

**Comment by juliovaz** [comment]
Just checked this and the plug-in that defines the doctypes is correct. I
even added a requires statement to the second plug-in that's using the
doctypes and it still fails as before. Do I need to include the catalog and
doctypes with the second plug-in? Never had to before.

**Comment by robander** [comment]
Doctypes don't have to be referenced from the second plugin - they only need to be referenced once.

**Comment by robander** [comment]
You showed a screen cap that has the `<rewriteURI>` tag. Do you also have something like this, later in the catalog?
```
<nextCatalog catalog="plugins/com.cook.doctypes/catalog-dita.xml" /> 
```

**Comment by juliovaz** [comment]
I'm truly at a loss here. I didn't expect the plug-in to work flawlessly
but I expected it to do something when I copied it to 2.0.1. I'll try
removing and reinstalling.

**Comment by robander** [comment]
I'm not sure why it would not work. I have a plugin that is nothing but doctypes (the IBM plugin mentioned earlier) - it works in 1.8.5 and 2.0.1 without any changes. So, I would expect yours to work in both as well.

**Comment by juliovaz** [comment]
There was a <nextCatalog> in the DITA catalog. I had other things to do but I'm going to try reinstalling the plugins again. I'll report back in a bit.

**Comment by juliovaz** [comment]
Tried reinstalling the plugins but still no luck. catalog and plugin looks fine. Not sure why it's not working. The nextcatalog statement was right in the catalog-dita.xml file so the DTDs should resolve. I'll have to put this aside for a bit and hope that some inspiration hits me but I would have thought it would just work.

**Comment by drmacro** [comment]
One thing to try is to manually following the catalog chain in oXygen using ctrl+enter and search for the PUBLIC value in the file it should be found in. That should be an accurate recreation of what the catalog resolver is doing and should reveal any typos that would lead to this sort of failure.

**Comment by juliovaz** [comment]
I understand what you're saying but without any changes to the plug-in, why
would just migrating cause this problem?

**Comment by juliovaz** [comment]
Finally found the typo. Looks like OT < 1.8.5 copied the DTD declarations into catalog-dita.xml but 2.0.1 doesn't. That meant that a typo I had in the xmlns for the catalog that wasn't detected in the earlier releases became a killer in 2.0.1. I share this so others who might run into it examine there xmlns declaration closely.

**Comment by robander** [comment]
Closing this - the specific issue was resolved, and there is a duplicate issue open #2224 to improve the DOTX008E error message.

**Comment by surighanta** [comment]
@juliovaz : What was the exact problem ? I mean in which file ? We are still fighting with it.
